From 783f632ab887a55c33fdc694082a6137d1a24c1e Mon Sep 17 00:00:00 2001
From: jkreierh <jkreierh@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 19 Jan 2017 09:32:59 +0000
Subject: [PATCH 0821/1076] MANTIS 44373 massaiCfgXB.exe version 1.6.5 - MANTIS
 44373: Config-error files format - The config files were written in "cooked"
 mode. This lead to additional newline characters. Now they are written
 "binary".

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1146 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/cpp/CfgXMgrB/inc/CfgXMgrB_def.h         | 14 ++++++++++++--
 .../Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h      | 14 ++++++++++++--
 .../Massai/cpp/CfgXMgrB/src/cfgEntity.cpp          |  2 +-
 3 files changed, 25 insertions(+), 5 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/CfgXMgrB_def.h b/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/CfgXMgrB_def.h
index 856e255c..a40546c1 100644
--- a/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/CfgXMgrB_def.h
+++ b/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/CfgXMgrB_def.h
@@ -495,8 +495,18 @@
 //! - Use an additional configuration backup, written immediately after
 //!   successful read.
 //! - Cleanup code, remove unnecessary compiler options.
-#define FILE_VER_BIN 1,6,4,0
-#define FILE_VER_DTS "1.6.4.0"
+// #define FILE_VER_BIN 1,6,4,0
+// #define FILE_VER_DTS "1.6.4.0"
+//
+//! @version 1.6.5
+//! @date    19.01.2017
+//! @author  JK
+//! @par     Change:
+//! - MANTIS 44373: Config-error files format
+//! - The config files were written in "cooked" mode. This lead
+//!   to additional newline characters. Now they are written "binary".
+#define FILE_VER_BIN 1,6,5,0
+#define FILE_VER_DTS "1.6.5.0"
 //
 
 
diff --git a/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h b/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h
index 156e5ae6..37663f0b 100644
--- a/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h
+++ b/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h
@@ -486,8 +486,18 @@
 //! - Use an additional configuration backup, written immediately after
 //!   successful read.
 //! - Cleanup code, remove unnecessary compiler options.
-#define FILE_VER_BIN 1,6,4,0
-#define FILE_VER_DTS "1.6.4.0"
+// #define FILE_VER_BIN 1,6,4,0
+// #define FILE_VER_DTS "1.6.4.0"
+//
+//! @version 1.6.5
+//! @date    19.01.2017
+//! @author  JK
+//! @par     Change:
+//! - MANTIS 44373: Config-error files format
+//! - The config files were written in "cooked" mode. This lead
+//!   to additional newline characters. Now they are written "binary".
+#define FILE_VER_BIN 1,6,5,0
+#define FILE_VER_DTS "1.6.5.0"
 //
 
 //-------------------------------------------------------------------------
diff --git a/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/cfgEntity.cpp b/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/cfgEntity.cpp
index 12ad94e3..84cd0d63 100644
--- a/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/cfgEntity.cpp
+++ b/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/cfgEntity.cpp
@@ -980,7 +980,7 @@ namespace
          mInFile<std::string> const &   FileData
       )
    {
-      std::ofstream OS( Filename.c_str() );
+      std::ofstream OS( Filename.c_str(), std::ios::binary );
       OS << FileData.container() << std::flush;
    }
 
-- 
2.41.0.windows.1

